broadinstitute/single_cell_portal_core
# Refine image pipeline: limit to top genes, generalize plot scale (SCP-5399)

This fine-tunes the image pipeline and related frontend code.  It improves reliability, relevance, and tailoring.

Previously (#1683), image pipeline attempted to cache images for _all_ genes.  This is expensive and error-prone.  To improve the reliability of pipeline runs while ensuring that the most important genes in a study show fast expression plots, image pipeline now only generates images for the top 100 genes per https://github.com/broadinstitute/scp-ingest-pipeline/pull/334.  

Ranges for expression, and x and y coordinates, are now tailored to the specific gene.  Before, hardcoded values were used.  Miscellaneous updates, like changing to use a Viridis color scale, are also now in place.

To test:
* Complete steps in https://github.com/broadinstitute/scp-ingest-pipeline/pull/334
* In line 236 of `image-pipeline/expression-scatter-plots.js`, set `const useDataCache = false`
* `cd image-pipeline`
* Run `touch log.txt`
* `yarn install`
* Run:
> IS_LOCAL=1 NODE_TLS_REJECT_UNAUTHORIZED=0 node expression-scatter-plots.js --accession="SCP367" --environment="staging" --bucket="fc-1e02d505-cdc3-4ac2-aa7f-13e2a6d79db6" --cluster="KaZhouAll_mt10_SCP_UMAP_coords_meta.csv" --cores="2"; tput bel
* Confirm success message at end of run

This satisfies SCP-5399.
